bin/flutter

```diff
@@ -93,7 +93,7 @@ if ! hash git 2>/dev/null; then
   exit 1
 fi
 # Test if the flutter directory is a git clone (otherwise git rev-parse HEAD would fail)
-if [ ! -d "$FLUTTER_ROOT/.git" ]; then
+if [ ! -e "$FLUTTER_ROOT/.git" ]; then
```

Comment:
I did the extremely dumb change, but maybe the better fix is:

```sh
if [ "$FLUTTER_ROOT/.git" == $( git rev-parse --show-toplevel ) ]; then
```

Reply:
Since this bit of the code isn't guaranteed to be run from within `$FLUTTER_ROOT` you'd need a `cd "$FLUTTER_ROOT"` in the subshell command, similar to what's on line 42. I think using a `-e` check is sufficient as a reasonable heuristic here (and is equivalent to what we do on Windows [here](https://github.com/flutter/flutter/blob/master/bin/flutter.bat#L39).